lib/legacy-logging.js

```diff
@@ -0,0 +1,764 @@
+    (_.isObject(logspec) && !_.keys(logspec).length)) {
+    map = [{ level: 'info+', handler: 'print' }]
+  }
+  else if (_.isString(logspec)) {
+    map = [logspec]
+  }
+  else if (_.isArray(logspec)) {
+    map = logspec
+  }
+  else if (_.isObject(logspec)) {
+    map = logspec.map ? logspec.map : [logspec]
+  }
+
+  var logrouter = new Patrun()
+
+  _.each(map, function (entry) {
+    if (_.isString(entry)) {
+      var entries = Legacy.logging_shortcut(entry)
+      entries.forEach(function (entry) {
+        makelogroute(entry, logrouter)
+      })
+    }
+    else if (entry) {
+      makelogroute(entry, logrouter)
+    }
+  })
+
+  return logrouter
+}
+
+function make_regex_handler (regex, handler) {
+  if (!_.isRegExp(regex)) {
+    var re_str = '' + regex
+    var re_flags = ''
+    var rere = /^\/(.*)\/([im]?)$/.exec(re_str)
+    if (rere) {
+      re_str = rere[1]
+      re_flags = rere[2]
+    }
+    regex = new RegExp(re_str, re_flags)
+  }
+
+  return function () {
+    var pretty = handlers.pretty.apply(null, arrayify(arguments)).join('\t')
+    if (regex.test(pretty)) {
+      return handler.apply(this, arguments)
+    }
+  }
+}
+
+function make_act_handler (act, handler) {
+  return function () {
+    if (('' + arguments[ log_index.act ]).indexOf(act) !== -1) {
+      return handler.apply(this, arguments)
+    }
+  }
+}
+
+function make_pin_handler (pin, handler) {
+  pin = _.isObject(pin) ? pin : Jsonic(pin)
+  var pinstr = Common.argpattern(pin)
+
+  return function () {
+    if (('' + arguments[ log_index.pin ]).indexOf(pinstr) !== -1) {
+      return handler.apply(this, arguments)
+    }
+  }
+}
+
+
+var makelogfuncs = legacy_logging.makelogfuncs = function (target) {
+  function makelogger (level) {
+    return function () {
+      var args = arrayify(arguments)
+      args.unshift(level)
+      target.log.apply(target, args)
+    }
+  }
+
+  target.log.debug = makelogger('debug')
+  target.log.info = makelogger('info')
+  target.log.warn = makelogger('warn')
+  target.log.error = makelogger('error')
+  target.log.fatal = makelogger('fatal')
+}
+
+legacy_logging.makelog = function (logspec, ctxt) {
+  var identifier = ctxt.id
+  var short = ctxt.short || logspec.short
+  var logrouter = makelogrouter(logspec)
+
+  var log = function (level, type) {
+    var when = new Date()
+    when.short$ = short
+
+    var args = arrayify(arguments).slice(2)
+
+    args.unshift(type)
+    args.unshift(level)
+    args.unshift(identifier)
+    args.unshift(when)
+
+    var pluginref = args[ log_index.plugin ]
+    pluginref = _.isString(pluginref) ? pluginref.trim() : pluginref
+
+    var routing = {
+      level: args[ log_index.level ],
+      type: args[ log_index.type ],
+      plugin: pluginref,
+      case: args[ log_index.case ]
+    }
+
+    var handler = logrouter.find(routing)
+
+    if (handler) {
+      var lastval = args[args.length - 1]
+      if (_.isFunction(lastval)) {
+        var logvals = []
+        try {
```

Comment:
hmmmmm, I know it's the legacy logging but I feel we should start to devolve these catches and just let it crash.